### ISSUE TITLE
"Replace selection" should work on all matches in a selection

### ISSUE THREAD

**Issue by humdingerb** [Genio-The-Haiku-IDE/Genio#287]
Currently it seems like the "Replace selection" action only works if the selection is exactly the search term.

I think it would be much more useful if it were "Replace in selection", i.e. you select several lines of code and the replace action would replace every occurance of the search term within that selection.

Maybe that was the intention from the beginnging, then this is a bug (and the tooltip should say "Replace in selection".

**Comment by Freaxed** [comment]
It's not a bug the code is checking if the selected text is exatrly the text to replace. Not very useful. 
To me the right solution should be to remove that action, add a checkbox at the right and call it "In selection". All the actions then will be relative to the selection only.

**Comment by humdingerb** [comment]
Agreed. Normally, one wouldn't even need the checkmark. What else would anyone expect when having a selection while replacing stuff? OTOH, the checkmark is an obvious hint how things work. On the still other hand, the checkmark means the user always has to make one more click in addition to the selection.

Is it intuitive enough? You guys decide... :)
